src/core/public/application/application_service.tsx

```diff
@@ -234,13 +235,20 @@ export class ApplicationService {
 
     const navigateToApp: InternalApplicationStart['navigateToApp'] = async (
       appId,
-      { deepLinkId, path, state, replace = false, openInNewTab = false }: NavigateToAppOptions = {}
+      {
+        deepLinkId,
+        path,
+        state,
+        replace = false,
+        openInNewTab = false,
+        skipAppLeave,
+      }: NavigateToAppOptions = {}
     ) => {
       const currentAppId = this.currentAppId$.value;
       const navigatingToSameApp = currentAppId === appId;
       const shouldNavigate = navigatingToSameApp
         ? true
-        : await this.shouldNavigate(overlays, appId);
+        : await this.shouldNavigate(overlays, appId, !!skipAppLeave);
```

Comment:
I wanted to be very specific about what behavior the flag influences and saw no harm in touching`shouldNavigate` since it's a private method. Either way works though and if the code paths are self-explanatory enough, that's fine with me too.

Reply:
This was more of a nit, I don't feel strongly about it -- mostly it just seemed like we could save ourselves from calling an async function if we already know `skipAppLeave === true`

Reply:
Skipping an async function makes sense: https://github.com/elastic/kibana/pull/128671/files#diff-94b600f79770d5081137b8938222dfc2718295a199b0685cd3fea4040114f451R249-R250